code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -715,9 +715,30 @@
 	if(!try_brew())
 		return
 	operate_for(brew_time)
-	coffeepot.reagents.add_reagent_list(list(/datum/reagent/consumable/coffee = 120))
-	coffee.Cut(1,2) //remove the first item from the list
+
+	// create a reference bean reagent list
+	var/list/reference_bean_reagents = list()
+	var/obj/item/food/grown/coffee/reference_bean = new /obj/item/food/grown/coffee(src)
+	for(var/datum/reagent/ref_bean_reagent in reference_bean.reagents.reagent_list)
+		reference_bean_reagents += ref_bean_reagent.name
+
+	// add all the reagents from the coffee beans to the coffeepot (ommit the ones from the reference bean)
+	var/list/reagent_delta = list()
+	var/obj/item/food/grown/coffee/bean = coffee[coffee_amount]
+	for(var/datum/reagent/substance in bean.reagents.reagent_list)
+		if(!(reference_bean_reagents.Find(substance.name)))	// we only add the reagent if it's a non-standard for coffee beans
```

Comment:
Why do we not want to include the reagents in a standard coffee bean? What are the implications?

Reply:
All the coffee obtained elsewhere (vendomats, the normal coffeemaker etc.) does follow this composition (just coffee). I figured it would be best to just keep it that way. Otherwise the coffee would include `Vitamin`, `Nutrient`, and `Coffeepowder` which I think would only add clutter

Reply:
Fair enough. Ideally all the standard reagents inside a coffee bean should be converted into coffee and not omitted like this but there is no clean way of doing that. You can resolve this